oak/server/rust/oak_runtime/src/channel.rs

```diff
@@ -0,0 +1,227 @@
+use oak_abi::OakStatus;
+
+use alloc::collections::VecDeque;
+use alloc::sync::Arc;
+use alloc::vec::Vec;
+
+use core::sync::atomic::AtomicUsize;
+use core::sync::atomic::Ordering::SeqCst;
+
+use crate::OakMessage;
+use crate::RwLock;
+
+type Messages = VecDeque<OakMessage>;
+
+/// TODO: doc
+struct Channel {
+    readers: AtomicUsize,
```

Comment:
I see this similar to a kernel owning processes and any other data structure that they need, e.g. file descriptors, threads, etc. To me it makes sense for the runtime to own everything, so it can mediate the creation and transfer of such entities (e.g. channels) between nodes, and can reclaim them when they are not needed any more. Pushing them down one level so that they are owned by the nodes, which in turn are owned by the runtime, would work too, but seems more problematic, especially if / when they need to be moved around. But I probably need to think more about it.